src/App.js

```diff
@@ -0,0 +1,175 @@
+import './App.css';
+import React from 'react';
+import GameField from './GameField';
+import Settings from './Settings';
+import Play from './Play';
+import Intro from './Intro';
+import rs_school_js from './rs_school_js.svg';
+import Timer from './Timer';
+import swapNumbers from './swapNumbers.js';
+import defineMoves from './defineMoves.js';
+import clickSound from './clickSound';
+import { VscGithubAlt } from "react-icons/vsc";
+import playWin from "./playWin.js";
+
+// import getLocalStorage from './getLocalStorage';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);    
+    this.updateSettings = this.updateSettings.bind(this);
+    this.handleStart = this.handleStart.bind(this);
+    this.handleSwap = this.handleSwap.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+    this.handleDemo = this.handleDemo.bind(this);
+    this.state = {
+      modalStyle: {display: 'block'},
+      sound: 'off',
+      mode: 'default',
+      stats: 'counts',
+      theme: 'light',
+      numbers: [1, 2, 3, 4, 5, 6, 7, 8, null],
+      toWin: [],
+      autoplay: false,
+      time: 0,
+      movesMade: 0,
+      moves: [],
+      gamesPlayed: 0,
+      gameLevel: 0,
+    }
+  }
+
+  // componentDidMount() {
+  //   getLocalStorage && this.setState(getLocalStorage);    
+  // }
+
+
+  updateSettings(settings) {
+    this.setState(settings);
+    this.setState({
+      numbers: [1, 2, 3, 4, 5, 6, 7, 8, null],
+      moves: [],
+      movesMade: 0,
+      toWin: [],
+      gamesPlayed: 0,
+    });
+  }
+
+  handleStart(newNumbers) {
+    this.setState({
+      numbers: newNumbers,
+      movesMade: 0,
+      toWin: [1, 2, 3, 4, 5, 6, 7, 8, null],
+      moves: defineMoves(newNumbers),
+      gamesPlayed: this.state.gamesPlayed + 1,
+      modalStyle: {display: 'block'},
+    });
+  }
+
+  handleDemo(demoNumbers) {
+    this.setState({
+      numbers: demoNumbers[0],
+      possibleClicks: [],
+      movesMade: 0,
+      toWin: [1, 2, 3, 4, 5, 6, 7, 8, null],
+    });
+
+    let demoSteps = demoNumbers[1];
+
+    demoSteps.forEach((step, i) => {
+      setTimeout(() => {
+        this.handleSwap(step);
+      }, (i+1) * 1000);
+    });
+  }
+
+  handleSwap(number) {
+    let nums = this.state.numbers;
+    let toWin = this.state.toWin;
+    this.state.sound === 'on' && clickSound('valid');
```

Comment:
It's better to use an IF-statement for better readability in a plain code. Use boolean expressions while assigning a value/returning a value from function/in JSX syntax/etc.